### PR TITLE
Allow writing of all properties.

### PR DESCRIPTION
**Pull request by jverkoey**

### CODE REVIEW

File: src/MDMAnimationTraits.h [hunk 1]
```diff
@@ -103,22 +103,22 @@
...
  The amount of time, in seconds, over which this animation should interpolate between its values.
  */
-@property(nonatomic, assign, readonly) NSTimeInterval duration;
+@property(nonatomic, assign) NSTimeInterval duration;
 
 /**
  The velocity and acceleration of the animation over time.
  */
-@property(nonatomic, strong, nullable, readonly) id<MDMTimingCurve> timingCurve;
+@property(nonatomic, strong, nullable) id<MDMTimingCurve> timingCurve;
```
Comment (romoore) [inline src/MDMAnimationTraits.h]: Can you describe the behavior/value if `nil`?

File: src/MDMAnimationTraits.h [hunk 2]
```diff
@@ -103,22 +103,22 @@
...
 
 /**
  The repetition characteristics of the animation.
  */
-@property(nonatomic, strong, nullable, readonly) id<MDMRepetitionTraits> repetition;
+@property(nonatomic, strong, nullable) id<MDMRepetitionTraits> repetition;
```
Comment (romoore) [inline src/MDMAnimationTraits.h]: Same request to describe `nil` behavior.

File: src/MDMSpringTimingCurve.m [hunk 3]
```diff
@@ -30,8 +30,7 @@ @implementation MDMSpringTimingCurve {
 @synthesize tension = _tension;
 
 - (instancetype)init {
-  [self doesNotRecognizeSelector:_cmd];
-  return nil;
+  return [self initWithMass:1 tension:341.023716023995 friction:36.9336548976131];
```
Comment (romoore) [inline src/MDMSpringTimingCurve.m]: Please cast these double literals to (CGFloat) for 32-bit arch support.

**Review by romoore — APPROVED** [review]
Please address comments before merging.

File: src/MDMAnimationTraits.h (hunk 1 above)
Comment (jverkoey) [reply src/MDMAnimationTraits.h]: Done.

File: src/MDMAnimationTraits.h (hunk 2 above)
Comment (jverkoey) [reply src/MDMAnimationTraits.h]: Done.

File: src/MDMSpringTimingCurve.m (hunk 3 above)
Comment (jverkoey) [reply src/MDMSpringTimingCurve.m]: Done.

**Pull request merged**